runelite-client/src/main/java/net/runelite/client/plugins/statusorbs/StatusOrbsOverlay.java

```diff
@@ -222,4 +265,61 @@ private void renderRegen(Graphics2D g, WidgetInfo widgetInfo, double percent, Co
 		g.setColor(color);
 		g.draw(arc);
 	}
+
+	public void updateSpec()
+	{
+		//0.2% spec restore per tick
+		if (client.getItemContainer(InventoryID.EQUIPMENT) == null)
+		{
+			return;
+		}
+		currentSpec = Math.floor((client.getVar(VarPlayer.SPECIAL_ATTACK_PERCENT) / 10) + (plugin.getSpecialPercentage() * 10));
+		int specTarget = 100;
+		final Item[] items = client.getItemContainer(InventoryID.EQUIPMENT).getItems();
```

Comment:
It would be better to use itemContainerChanged and cache this, so you don't have to get all items every tick